### ISSUE TITLE
trouble with input arguments

### ISSUE THREAD

**Issue by yozman** [composer/composer#5855]
I have a script like this
```json
"scripts": {
  "serve": "php -S localhost:9000 -t public scripts/serve.php"
}
```

I want to use like
`composer serve aomg`

its output is like
```shell
> php -S localhost:9000 -t public scripts/serve.php 'aomg'

```

so how can I get the string `aomg` in my `scripts/serve.php`?
could you please tell me the right way?

**Comment by alcohol** [comment]
We do not support arguments for the scripts section when using this approach.

**Comment by Seldaek** [comment]
We do support it, as it is even shown in the output of the command being executed. In scripts/serve.php you should be able to access `$_SERVER['argv']` to find the aomg argument. The problem is that when you run the `php -S` built-in web server argv is not accessible it seems, so in that case your only option is to pass in environment variables like `FOO=aomg composer serve` and then retrieve it in the serve.php as `$_SERVER['FOO']`. Except it seems that also doesn't work.. because by default it doesn't allow environment vars in the built-in web server.

If you run it like this:

```
"scripts": {
  "serve": "php -d variables_order=EGPCS -S localhost:9000 -t public scripts/serve.php"
}
```

Then it allows the `$_ENV` var registration, and by running `FOO=aomg composer serve` you can then access `$_ENV['FOO']`.

In other words, composer does this fine but the built-in web server has severe limitations.

**Comment by yozman** [comment]
@alcohol @Seldaek 
could you please add feature
like `@composer`,
for example `@input` or sth.

**Comment by Seldaek** [comment]
I am not sure what you are asking for there, the issue is really php's web server, not what composer does.

**Comment by yozman** [comment]
@Seldaek
maybe there should be some string to define before
like
```json
"scripts": {
  "serve": "FOO=@argv php -S localhost:9000 -t public scripts/serve.php"
}
```

then I run `php serve aomg`
the real command is
`FOO='aomg' php -S localhost:9000 -t public scripts/serve.php`

**Comment by alcohol** [comment]
I would suggest you look at [`phpdotenv`](https://github.com/vlucas/phpdotenv) instead.
